copr_builder/srpm_builder.py

```diff
@@ -147,20 +147,20 @@ def _make_srpm(self, archive):
         git_dir = self.git_repo.gitdir
         pkg_name = self.project_data[PACKAGE_CONF]
 
-        # copy source archive to rpmbuild/SOURCES
-        ret, rpmsource = run_command('rpm -E %_sourcedir')
-        if ret != 0:
-            raise SRPMBuilderError('Failed to get rpmbuild SOURCE directory location.')
-
-        shutil.copyfile('%s/%s' % (git_dir, archive),
-                        '%s/%s' % (rpmsource, archive))
+        # create 'packaging' directory in gitdir
+        rpmdir = os.path.join(self.git_dir, 'packaging')
+        if not os.path.exists(rpmdir):
+            os.mkdir(rpmdir)
 
         # build the srpm
-        command = 'rpmbuild -bs %s' % self.spec_file
+        data = {'srcdir': self.git_dir, 'rpmdir': rpmdir, 'spec': self.spec_file}
+        command = 'rpmbuild -bs --define "_sourcedir {srcdir}" --define "_specdir {rpmdir}"' \
+                  ' --define "_builddir {rpmdir}" --define "_srcrpmdir {rpmdir}"' \
+                  ' --define "_rpmdir {rpmdir}" {spec}'.format(**data)
```

Comment:
This seems to me little bit fragile. Wouldn't it be better to replace all this stuff with just:
```
mock -r fedora-rawhide-x86_64 --buildsrpm --spec {spec} --sources {tarball}
```

However, I'm not sure if you have the tarball somewhere.